src/_guides/language/effective-dart/usage.md

```diff
@@ -245,87 +245,66 @@ void error([String? message = null]) {
 }
 {% endprettify %}
 
+<a id="prefer-using--to-convert-null-to-a-boolean-value"></a>
+### DON'T use `true` or `false` in equality operations
 
-### PREFER using `??` to convert `null` to a boolean value.
-
-This rule applies when an expression can evaluate to `true`, `false`, or `null`,
-and you need to pass the result to something that expects a non-nullable boolean
-value. A common case is using the result of a null-aware method call as a
-condition. You can "convert" `null` to `true` or `false` using `==`, but we
-recommend using `??`:
+Using the equality operator to evaluate a *non-nullable* Boolean expression 
+against a Boolean literal is redundant. 
+It's always simpler to eliminate the equality operator, and add `!` if necessary:
```

Comment:
This mention of adding "`!` if necessary" is a bit confusing to me, because it makes it seem like it will be part of the solution, but the guideline goes on to say "you should only use `??` or an explicit `!= null` check." and doesn't mention `!`. 

Depending on the goal of this mention, perhaps just remove the mention of `!` as I think the other two solutions are generally a better choice anyway.

Reply:
That's meant to capture the case of negating the expression, depending on what you're trying to accomplish. These are both "good" for non-nullable boolean expressions:
```
if (nonNullableBool) { ... }

if (!nonNullableBool) { ... }
```

So maybe removing ` and add `!` if necessary` and just adding the second "good" example is clearer. Or even keeping the clause and adding the second example, that'd probably be clearest

Reply:
Oh my, sorry! I was so focused on nullability, I thought this was talking about the null assertion operator and was pretty confused.

I think adding a second example would be great, but I'd love to also modify the clause so people don't read the text incorrectly as being about the null assertion operator. 

```suggestion
It's always simpler to eliminate the equality operator, 
and use the unary negation operator (`!`) if necessary:
```